File: code/model/comment.py

from sparrow import *

from .owentity import *
from .status import Status
from .user import User

class Comment(RTOwEntity):
    key = CID = KeyProperty()
    status = RTReference(Status)
    author = RTReference(User)
    date = Property(int)
    date_edited = Property(int)  # If they are the same, no edits
    text = Property(str)

    async def can_delete(self, usr_uid, db):
        u = await User.find_by_key(self.author, db)
        return u.key == usr_uid

File: code/js/prototypes/comment.js
Comment.prototype = new DataType();
Comment.prototype.constructor = Comment;
Comment.prototype._key = ["CID", "SID"];

function Status(CID, date, date_edited, author_UID, SID, text) {
	this.CID = CID;
	this.date = date;
	this.date_edited = date_edited;
	this.author_UID = author_UID;
	this.SID = SID;
	this.text = text
}
